examples/experimental/horizontal-pod-autoscaling/README.md

```diff
@@ -0,0 +1,207 @@
+and experimental controllers turned on in controller manager (``--enable-experimental=true``).
+This can be simply achieved on GCE by exporting ``KUBE_ENABLE_EXPERIMENTAL_API=true`` before running ```kube-up.sh``` script.
+
+The required version of kubectl is also 1.1.
+
+
+## Step One: Run & expose php-apache server
+
+To demonstrate horizontal pod autoscaler we will use a custom docker image based on php-apache server.
+The image can be found [here](image/).
+It defines [index.php](image/index.php) page which performs some CPU intensive computations.
+
+First, we will start a replication controller running the image and expose it as an external service:
+```console
+$ kubectl create -f ./examples/experimental/horizontal-pod-autoscaling/rc-php-apache.yaml
+replicationcontrollers/php-apache
+
+$ kubectl expose rc php-apache --port=80 --type=LoadBalancer
+NAME         LABELS           SELECTOR         IP(S)     PORT(S)
+php-apache   run=php-apache   run=php-apache             80/TCP
+```
+
+Now, we will wait some time and verify that both the replication controller and the service were correctly created and are running. We will also determine the IP address of the service:
+
+```console
+$ kubectl get pods
+NAME               READY     STATUS    RESTARTS   AGE
+php-apache-wa3t1   1/1       Running   0          12m
+
+$ kubectl describe services php-apache | grep "LoadBalancer Ingress"
+LoadBalancer Ingress:	146.148.24.244
+```
+
+We may now check that php-apache server works correctly by calling ``curl`` with the service's IP:
+
+```console
+$ curl http://146.148.24.244
+OK!
+```
+
+Please notice that when exposing the service we assumed that our cluster runs on a provider which supports load balancers (e.g.: on GCE).
+If load balancers are not supported (e.g.: on Vagrant), we can expose php-apache service as ``ClusterIP`` and connect to it using the proxy on the master:
+
+```console
+$ kubectl expose rc php-apache --port=80 --type=ClusterIP
+NAME         LABELS           SELECTOR         IP(S)     PORT(S)
+php-apache   run=php-apache   run=php-apache             80/TCP
+
+$ kubectl cluster-info | grep master
+Kubernetes master is running at https://146.148.6.215
+
+$ curl -k -u admin:password https://146.148.6.215/api/v1/proxy/namespaces/default/services/php-apache/
+OK!
+```
+
+
+## Step Two: Create horizontal pod autoscaler
+
+Now that the server is running, we will create a horizontal pod autoscaler for it.
+To create it, we will use [hpa-php-apache.yaml](hpa-php-apache.yaml) file, which look like this:
+
+```yaml
+apiVersion: experimental/v1alpha1
+kind: HorizontalPodAutoscaler
+metadata:
+  name: php-apache
+  namespace: default
+spec:
+  maxReplicas: 10
+  minReplicas: 1
+  scaleRef:
+    kind: ReplicationController
+    name: php-apache
+    namespace: default
+  target:
+    quantity: 100m
+    resource: cpu
+```
+
```

Comment:
I would add a sentence here explaining what this does, something like "This will create a horizontal pod autoscaler that maintains between 1 and 10 replicas of the Pods controlled by the `php-apache` replication controller you created in the first step of these instructions. Roughly speaking, the horizontal autoscaler will increase and decrease the number of replicas (via the replication controller) so as to maintain an average CPU utilization across all Pods of 100 millicores. See \<reference to proposals/ doc here\> for more details on the algorithm."